### DISCUSSION TITLE
Does PyMC marketing allows for geo-hierarchical modelling?

### DISCUSSION THREAD

**Discussion by fredxu-8451** [pymc-labs/pymc-marketing#648] [General]
the example I walked through was national as a whole. I wonder if pymc marketing can do geo-hierarchical specification to enrich the data input as meridian did?

**Comment by williambdean** [comment]
Hi,
There is not currently a geo-hierarchical model that is implemented right out of the box. I would like to highlight that the adstock and saturation transformation do support additional dimensions so hierarchical models can be implemented with a bit of pymc knowledge. 

There are some existing requests for this functionality so it could be part of a future feature. 
If you provide some additional details, then we can be sure to take into thought! 

Let me know if you have any follow ups

**Reply by fredxu-8451** [reply]
Thank you Will for your quick reply. I am new to pymc and pymc marketing. I wonder if you or your team have any transferable knowledge (example would be better) to cultivate pymc components to customize for my purpose. we are looking for a bayesian MMM that can do geo-hierarchical specification, allow time variant media channels, incorporate reach and frequency and flexible prior customization. 
Sorry I asked a lot, but super excited to learn more about pymc and pymc marketing

**Reply by cetagostini** [reply]
Hello @fredxu-8451 

If you need a quick guidance about how to implement the model into the PyMC framework, you could check the two **DRAFT PRs** we had previously month.

- #599 
- https://github.com/pymc-labs/pymc-marketing/pull/460

As @wd60622 mentioned, we haven't yet decided how to quickly bring these ideas into the PyMC-Marketing API but you can still follow these references to adjust your model.

Hope it helps!

**Reply by williambdean** [reply]
Hierarchical parameters are supported with the dictionary configurations of 0.7.0 and upcoming Prior class both of which integrate into all media transformation as well as fourier seasonality
Here is the latest example notebook: https://www.pymc-marketing.io/en/latest/notebooks/mmm/mmm_components.html#example-custom-mmm

**Reply by fredxu-8451** [reply]
This is magnificent. I just discussed with my coworkers last week about your new release. I think it is time to schedule a meeting to discuss a possible solution to bring pymc-marketing in house. Can I just use that [free 30-minute strategy session] to get it started?

**Reply by williambdean** [reply]
> This is magnificent. I just discussed with my coworkers last week about your new release. I think it is time to schedule a meeting to discuss a possible solution to bring pymc-marketing in house. Can I just use that [free 30-minute strategy session] to get it started?

Hi @fredxu-8451 
Feel free to setup a call with @nialloulton by using this link here: https://calendly.com/niall-oulton

**Comment by AlfredoJF** [comment]
Just wondering if adding extra dims like `geo` to the `MMM` class is in the works. The notebook components notebook https://www.pymc-marketing.io/en/latest/notebooks/mmm/mmm_components.html#example-custom-mmm is really helpful but as mentioned in there, it feels odd losing the in-built `pymc-marketing` functionalities.

> - scaling of input and output data
> - plotting methods for parameters, predictive data, contributions, etc
> - customized adstock and saturation transformations
> - out of sample predictions
> - lift test integration
> - budget optimization

**Comment by williambdean** [comment]
Give the new class from #1036 a try. Here is an example notebook https://www.pymc-marketing.io/en/stable/notebooks/mmm/mmm_multidimensional_example.html
